pkg/bitcoin/transaction.go

```diff
@@ -89,6 +93,60 @@ func (t *Transaction) Serialize(
 	}
 }
 
+// SerializeVersion serializes the transaction version to a little-endian
+// 4-byte array.
+func (t *Transaction) SerializeVersion() [4]byte {
+	result := [4]byte{}
+	binary.LittleEndian.PutUint32(result[:], uint32(t.Version))
+	return result
+}
+
```

Comment:
It would really help to have [the tables](https://github.com/keep-network/tbtc-v2/blob/031249d0001ab065e80a48e331da1155a636b9e8/solidity/contracts/bridge/BitcoinTx.sol#L27-L91) on the top of `type Transaction struct` and then for each `Serialize*` function denote which parts we are serializing.

For example, in `SerializeInputs` the output has `[ tx_in_count | tx_in]`.

Reply:
I was looking at the Solidity documentation for`BitcoinTx.Info` to understand why are we taking 4 bytes or why are we skipping 4 bytes in other functions. The references to `developer.bitcoin.org` are helpful but IMO they are not as helpful as having the structure described in the code comments as we have in `BitcoinTx.Info`. It is not a blocker, I can leave this conversation open to marinate for a while to see if we can find a way to port some of those docs to Go.

Reply:
Sure! What I want to avoid is putting too many docs as they add additional maintenance work and can obfuscate the picture even more. The Go `bitcoin` package differs from the Solidity code we were working with in the past. In Solidity, we used to deal with pure bytes quite often so a huge amount of docs was the only way to explain what happens. In Go `bitcoin` package, the code itself is more verbose. All Bitcoin primitives and their contents are represented by dedicated Go types which have their own documentation. In that context, having a huge block of docs above `bitcoin.Transaction` just feels a bit redundant.